beacon/types/exec_data_test.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	withdrawalsHash1 = common.HexToHash("dead")
+
+	isthumusExecData = engine.ExecutableData{
+		ParentHash:      common.HexToHash("parent"),
+		FeeRecipient:    common.HexToAddress("0x376c47978271565f56DEB45495afa69E59c16Ab2"),
+		StateRoot:       common.HexToHash("sRoot"),
+		ReceiptsRoot:    common.HexToHash("rRoot"),
+		LogsBloom:       common.Hex2Bytes("0x376c47978271565f56DEB45495afa69E59c16Ab2"),
+		Random:          common.HexToHash("randao"),
+		BaseFeePerGas:   hexutil.MustDecodeBig("0x2000000"),
+		Transactions:    [][]byte{},
+		Withdrawals:     []*types.Withdrawal{},
+		WithdrawalsRoot: &withdrawalsHash1,
+	}
+
+	executableDataSamples = []engine.ExecutableData{
+		isthumusExecData,
+	}
+)
+
+func TestExecutableDataJSONEncodeDecode(t *testing.T) {
```

Comment:
This could be good as a fuzz test

Reply:
As discussed with @tynes , will move this to a fuzz test in a follow up.

Reply:
Done - Moved `beacon.Header` and `types.xyz` encode/decode rountrip to a fuzz test.